modules/internal/ChapelBase.chpl

```diff
@@ -153,12 +153,17 @@ module ChapelBase {
   inline proc ==(param a: bool, param b: bool) param return __primitive("==", a, b);
   inline proc ==(param a: int(?w), param b: int(w)) param return __primitive("==", a, b);
   inline proc ==(param a: uint(?w), param b: uint(w)) param return __primitive("==", a, b);
+  //
   inline proc ==(param a: enumerated, param b: enumerated) param where (a.type == b.type) return __primitive("==", a, b);
+  //
+  // NOTE: Only '==' is implemented in the compiler on param enums
+  // (and only only on param enums of the same type), so all other
+  // param enum routines need to be defined as module code.
 
   inline proc !=(param a: bool, param b: bool) param return __primitive("!=", a, b);
   inline proc !=(param a: int(?w), param b: int(w)) param return __primitive("!=", a, b);
   inline proc !=(param a: uint(?w), param b: uint(w)) param return __primitive("!=", a, b);
-  inline proc !=(param a: enumerated, param b: enumerated) param where (a.type == b.type) return __primitive("!=", a, b);
+  inline proc !=(param a: enumerated, param b: enumerated) param where (a.type == b.type) return __primitive("!=", chpl__enumToOrder(a), chpl__enumToOrder(b));
```

Comment:
I would prefer to avoid using chpl__enumToOrder() here,
because the == case does not use those.
`!__primitive("==", a, b)` seems appropriate.

Reply:
The `==` case doesn't use them because it's implemented in the compiler as noted in the comment that follows it.  This case is symmetrical with all of the other comparison operators (which can't be implemented in terms of "=="), which seems preferable to me to introducing asymmetry just for this case.

Reply:
OK, that's fine, then.